packages/meditrak-app/app/assessment/QrCodeScreen.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Tupaia MediTrak
+ * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useRef } from 'react';
+import { connect } from 'react-redux';
+import { ScrollView, StyleSheet, View } from 'react-native';
+import Share from 'react-native-share';
+import RNFS from 'react-native-fs';
+import { FileSystem } from 'react-native-file-access';
+
+import { addMessage } from '../messages';
+import { Heading, TupaiaBackground, Button, Divider, QrCode } from '../widgets';
+import { THEME_COLOR_ONE, THEME_FONT_SIZE_ONE } from '../globalStyles';
+import { formatPlural } from '../utilities';
+
+const shareQrCode = (qrCodeRef, filename) => {
+  qrCodeRef.toDataURL(dataURL => {
+    const shareOptions = {
+      filename,
+      type: 'image/*',
+      url: `data:image/png;base64,${dataURL}`,
+    };
+    Share.open(shareOptions);
+  });
+};
+
+const downloadQrCode = (qrCodeRef, filename, onSuccess, onFail) => {
+  qrCodeRef.toDataURL(async dataURL => {
+    try {
+      const fullFilename = `${filename}.png`;
+      const filePath = `${RNFS.DocumentDirectoryPath}/${fullFilename}`;
+      await RNFS.writeFile(filePath, dataURL, 'base64');
+      await FileSystem.cpExternal(filePath, fullFilename, 'downloads');
+      onSuccess();
+    } catch (error) {
+      onFail(error.message);
+    }
+  });
+};
+
+const QrCodeScreenComponent = props => {
+  const { displayDownloadMessage } = props;
+  const { qrCodes = [], onClose = () => {} } = props.navigation.state.params;
+  const qrCodeImgRefs = useRef(new Array(qrCodes.length));
+
+  return (
+    <TupaiaBackground>
+      <Heading
+        text={`Share QR ${formatPlural('Code', 'Codes', qrCodes.length)}`}
+        style={localStyles.heading}
+      />
+      <ScrollView>
+        {qrCodes.map(({ data, name }, index) => (
+          <View key={data} style={localStyles.qrCodeContainer}>
+            <QrCode
+              getRef={ref => {
+                qrCodeImgRefs.current[index] = ref;
```

Comment:
Looks good. The only query I have is probably more semantics. But when you create qrCodeImgRefs, this is technically one ref right, with the current value an array. So is it correct to be referring to the element values as refs?

Reply:
Ah good point, technically it should be `qrCodesRef` since it's one ref to many codes 👍 I'll rename